fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/provider/impl/S3Provider.java

```diff
@@ -171,10 +171,16 @@ private static AmazonS3 getClient(boolean iam, String cosApiKeyProperty, String
      * @return
      */
     public boolean exists() {
-        boolean ex = client != null && client.doesBucketExistV2(bucketName);
-        if (ex == false) {
+        boolean ex = client != null;
+
+        // We only want to log a warning here, and assume it's true if the client exists
```

Comment:
So client will always be null if the bucket doesn't exist?

Reply:
Or, if not, can you help me understand how will the end-user-facing error will change from this commit (in the case of a non-existent bucket)?

Reply:
previously we were checking if the client was valid, and then doing the check.  In this case, we're only warning if the client doesn't exist.

Reply:
I fixed line 180 to include ex, it was suppose to be !ex

Reply:
I'm sorry, I'm still a bit confused.  The logic seems the same as the original logic (outside this PR) now.
I think maybe you were purposefully no longer relying on `client.doesBucketExistV2(bucketName)` and I was trying to understand why (and how that would bubble up in the errors we generate).

Reply:
ok, i had another look and i understand how its different now.